src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -81,34 +80,33 @@ public void deliveryComplete(IMqttDeliveryToken token) {
     };
 
     /**
-     * Ctr for MQTTClient.
+     * Construct an MQTTClient.
      *
-     * @param topics             topics passed in by Nucleus
+     * @param brokerUri          broker uri
+     * @param clientId           client id
      * @param mqttClientKeyStore KeyStore for MQTT Client
      * @param executorService    Executor service
      * @throws MQTTClientException if unable to create client for the mqtt broker
      */
     @Inject
-    public MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
+    public MQTTClient(URI brokerUri, String clientId, MQTTClientKeyStore mqttClientKeyStore,
                       ExecutorService executorService) throws MQTTClientException {
-        this(topics, mqttClientKeyStore, executorService, null);
+        this(brokerUri, clientId, mqttClientKeyStore, executorService, null);
         try {
             this.mqttClientInternal = new MqttClient(brokerUri.toString(), clientId, dataStore);
         } catch (MqttException e) {
             throw new MQTTClientException("Unable to create an MQTT client", e);
         }
     }
 
-    protected MQTTClient(Topics topics, MQTTClientKeyStore mqttClientKeyStore,
-                         ExecutorService executorService, IMqttClient mqttClient) throws MQTTClientException {
-        try {
-            this.brokerUri = BridgeConfig.getBrokerUri(topics);
-        } catch (URISyntaxException e) {
-            throw new MQTTClientException("Invalid brokerUri configuration", e);
-        }
+    protected MQTTClient(URI brokerUri, String clientId, MQTTClientKeyStore mqttClientKeyStore,
+                         ExecutorService executorService, IMqttClient mqttClient) {
+        Objects.requireNonNull(brokerUri, "Broker URI cannot be null");
+        Objects.requireNonNull(clientId, "Client ID cannot be null");
```

Comment:
Can it be null though? At this point in code, this has become a developer error. 
We should catch these Invalid input early (in MqttBridge) and surface them as customer errors.

Are these developer asserts at this point?

Reply:
You can use @NonNull annotations to enforce that. It will throw NPE which means a developer error.

Reply:
yep they're developer asserts, will adjust to use nonnull annotation 👍